### DISCUSSION TITLE
Synchronize users/passwords

### DISCUSSION THREAD

**Discussion by aducom** [e107inc/e107#4789] [Q&A Core]
I want to synchronize user id and password over the web applications that are on my website. Is there some kind of api that I can use to add / alter a user with his password? If not, how can I perform this task? I am using the standard E107 method, I could not find the key and salt for the logon. Point is that users start with a registration on E107.

**Comment by Alex-e107nl** [comment]
Hi Albert, 
when I need a script within e107 I use the code below so that it is only shown when a member is logged in to e107. Does that help or do you really want to use the login details of e107 in another application?
```
require_once("../class2.php");

if (!USER) {
   $ns->tablerender("FOUT!", "U moet ingelogd zijn om deze pagina te bekijken!");
  
   exit;
}
```

MvG
Alex

**Comment by aducom** [comment]
Hi Alex, long time...

I have several web applications running and I want to have a mechanism that I can synchronize users and passwords. Most applications have APIs so that I can add/maintain users (with an api key). But E107 is giving me trouble doing so. In some cases, users start by taking an account on E107. Then I can extract this data directly from the database, but the password cannot be used. Using a centralized auth (like OAuth) is not an option for now, unfortunately. So I have a separate application for maintaining which works fine, except for E107.

**Reply by Alex-e107nl** [reply]
Yeah it's toooooo long....

Have you looked at https://gitter.im/e107inc/e107?source=orgpage and search for password hash?
(login first)
